includes/Admin/Admin_AJAX.php

```diff
@@ -208,9 +209,39 @@ public function run_checks() {
 			wp_send_json_error( $valid_nonce, 403 );
 		}
 
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			$runner = new AJAX_Runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
+				500
+			);
+		}
+
+		$checks = isset( $_REQUEST['checks'] ) ? array_filter( $_REQUEST['checks'] ) : array();
+		$plugin = filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_STRING );
+
+		try {
+			$runner->set_check_slugs( $checks );
+			$runner->set_plugin( $plugin );
+			$results = $runner->run();
+		} catch ( Exception $error ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-request', $error->getMessage() ),
+				400
+			);
+		}
+
 		wp_send_json_success(
 			array(
-				'message' => __( 'Verified!', 'plugin-check' ),
+				'results' => $this->format_results( $results ),
```

Comment:
See my other comment: I don't think we need to format to cater for a specific opinionated output format, we can return the raw data here and then handle formatting as needed in JS, according to how we use the template.
```suggestion
				'errors    => $results->get_errors(),
				'warnings' => $results->get_warnings(),
```

Reply:
Thanks @felixarntz updated.